service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionService.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.fpl.service.casesubmission;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisCaseSubmission;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C110A;
+import static uk.gov.hmcts.reform.fpl.utils.SubmittedFormFilenameHelper.buildFileName;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseSubmissionService  {
+    private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final UploadDocumentService uploadDocumentService;
+    private final CaseSubmissionTemplateDataGenerationService documentGenerationService;
+    private final ObjectMapper mapper;
+
+    public Document generateSubmittedFormPDF(final CaseDetails caseDetails) {
```

Comment:
Thoughts on changing this to just take case data and the id? Would mean you don't need to the autowire in the object mapper as well

Reply:
Although just noticed that the buildFileName method requires it so I guess it is down to whether you want to refactor that as well